### DISCUSSION TITLE
language detection not working with nextjs 11 and next-i18n 13

### DISCUSSION THREAD

**Discussion by pedrohba1** [i18next/next-i18next#2037] [Q&A] [answered]
I have an application running on https://stadium.equalssport.com/

It has two languages, which run on the paths: https://stadium.equalssport.com/en and https://stadium.equalssport.com/pt

However, it does not auto detect the language from the browser. I  can't figure out why. I strongly believe it might be because I'm using nextjs 11, because we deploy it using serverless, so it only supports version 10 and 11.

for the page `/welcome` we use this to fetch translations:

```ts

export const getServerSideProps: GetServerSideProps = withSSRUserAuth(
  async ctx => {
    const { locale } = ctx;

    return {
      props: {
        ...(await serverSideTranslations(
          locale ?? "pt",
          ["welcome", "toast"],
          i18nextConfig
        )),
      },
    };
  }
);
```

And My `next-i18next.config.js` goes like this: 

```
// @ts-check

/**
 * @type {import('next-i18next').UserConfig}
 */
 module.exports = {
  // https://www.i18next.com/overview/configuration-options#logging
  // Restore this line to enable i18n debug logging
  debug: process.env.NODE_ENV === 'development',

  i18n: {
    defaultLocale: 'pt',
    locales: ['en', 'pt'],
  },
  localeDetection: true,

  /** To avoid issues when deploying to some paas (vercel...) */
  localePath:
    typeof window === 'undefined'
      ? require('path').resolve('./public/locales')
      : '/locales',
      returnObjects: true,
  reloadOnPrerender: process.env.NODE_ENV === 'development',
}
```

Versions of packages we are using: 

 ```
     "i18next": "^22.0.3",
    "next-i18next": "^13.0.0",
    "react-i18next": "^11.18.6",
    "next": "11.1.3",
 ```

Is there a way to make it handle language detection? I beleive it's possible to do something related to the `accepted-language` header but I'm not sure.

**Comment by adrai** [comment, marked as answer]
localeDetection is an option in the i18n object: https://nextjs.org/docs/advanced-features/i18n-routing#prefixing-the-default-locale
The whole language detection and routing topic is directly solved by next.js, there is no logic about that in next-i18next

**Reply by pedrohba1** [reply]
It's happening because I'm deploying my application using serverless I beleive. I will be closing it then.
